x-pack/plugins/apm/public/components/app/TransactionDetails/WaterfallWithSummmary/WaterfallContainer/Waterfall/waterfall_helpers/waterfall_helpers.ts

```diff
@@ -243,12 +244,14 @@ const getWaterfallItems = (items: TraceAPIResponse['trace']['items']) =>
  * @param waterfallItems
  */
 const reparentSpans = (waterfallItems: IWaterfallItem[]) => {
-  return waterfallItems.map(waterfallItem => {
+  // clones the waterfall items to avoid mutating the original
+  const clonedWaterfallItems = cloneDeep(waterfallItems);
+  return clonedWaterfallItems.map(waterfallItem => {
     if (waterfallItem.docType === 'span') {
       const childId = waterfallItem.doc.child?.id;
       if (childId) {
         childId.forEach(id => {
-          const item = waterfallItems.find(_item => _item.id === id);
+          const item = clonedWaterfallItems.find(_item => _item.id === id);
           if (item) {
             item.parentId = waterfallItem.id;
```

Comment:
This only changes the `parentId` but a `waterfallItem` also has a `parent` property. Should that be updated too?

![image](https://user-images.githubusercontent.com/209966/81548622-d8759100-937d-11ea-8eff-eb62348da048.png)

Reply:
It won't change anything in the waterfall, but I will change it to make it consistent.

Reply:
I'm wondering why we have both? Isn't it enough with just `parent`? Then it's easy to get parent id via `parent.id`?

Reply:
@sqren It's not needed since the function `getOrderedWaterfallItems` adds the parent into the waterfall item which was already reparented.

Reply:
> I'm wondering why we have both? Isn't it enough with just `parent`? Then it's easy to get parent id via `parent.id`?

Indeed that would make more sense since it's only used to group children and parents